app/models/roster_entry.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class RosterEntry < ApplicationRecord
+  belongs_to :roster
+  belongs_to :user, optional: true
+
+  validates :identifier, presence: true
+  validates :roster,     presence: true
```

Comment:
For some reason, not having the explicit validation makes the db throw an exception before the model